### ISSUE TITLE
Presigned URLs don't work for object with spaces or + in their names

### ISSUE THREAD

**Issue by p-himik** [localstack/localstack#3759]
# Type of request: This is a ...

- [x] bug report
- [ ] feature 

# Detailed description
Presigned URLs don't work for object with spaces or `+` in their names.
So e.g. when `test.txt` works just fine, `test 1.txt` and `test+1.txt` don't.

## Expected behavior
Issuing a GET request to a presigned URL should return the right object.

## Actual behavior
It returns the `SignatureDoesNotMatch` error.

# Steps to reproduce

## Command used to start LocalStack
Just the default Docker image, with `-e SERVICES=s3` and `-e DATA_DIR=localstack-data-dir`.

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
Notice that I'm using my own `awslocal` alias:
```bash
$ alias awslocal='env AWS_ACCESS_KEY_ID="test" AWS_SECRET_ACCESS_KEY="test" aws --endpoint-url=http://localhost:4566'
$ awslocal s3 mb s3://test
make_bucket: test
$ echo hello > test.txt
$ awslocal s3 cp test.txt s3://test/test.txt
upload: ./test.txt to s3://test/test.txt
$ curl `awslocal s3 presign s3://test/test.txt`
hello
$ awslocal s3 cp test.txt s3://test/test\ 1.txt
upload: ./test.txt to s3://test/test 1.txt
$ curl `awslocal s3 presign s3://test/test\ 1.txt`
<?xml version="1.0" encoding="utf-8"?>
<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided.                     Check your key and signing method.</Message><RequestId>761aa640</RequestId><HostId>a29d4446</HostId><AWSAccessKeyId>test</AWSAccessKeyId><StringToSign></StringToSign><SignatureProvided>af0c9cc819bd125eac617788e7a02779beae45919b04c0e7bc0cef6360514f32</SignatureProvided><StringToSignBytes>61663063396363383139626431323565616336313737383865376130323737396265616534353931396230346330653762633063656636333630353134663332</StringToSignBytes></Error>
$ awslocal s3 cp test.txt s3://test/test+1.txt
upload: ./test.txt to s3://test/test+1.txt
$ curl `awslocal s3 presign s3://test/test+1.txt`
<?xml version="1.0" encoding="utf-8"?>
<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided.                     Check your key and signing method.</Message><RequestId>2e99fbdb</RequestId><HostId>efe445f4</HostId><AWSAccessKeyId>test</AWSAccessKeyId><StringToSign></StringToSign><SignatureProvided>0e512b88a78e548f88d02058bbd26be831c593149fdb56b8328447773b813d9b</SignatureProvided><StringToSignBytes>30653531326238386137386535343866383864303230353862626432366265383331633539333134396664623536623833323834343737373362383133643962</StringToSignBytes></Error>
```

The same exact commands work just fine on real AWS S3.

**Comment by whummer** [comment]
Thanks for reporting @p-himik . This should (hopefully) be fixed in #3760 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks!

**Comment by p-himik** [comment]
Seems to work, thanks!
